pages/qr.tsx

```diff
@@ -1,14 +1,21 @@
-import * as React from 'react'
 import Head from 'next/head'
-
-import { Text, Card, Box } from '~ui/core'
+import * as React from 'react'
 import { isFormal } from '~lib/config'
 import { ArrowsLeft, ArrowsRight } from '~ui/anicons'
+import { Box, Card, Text } from '~ui/core'
 import { MobileApp } from '~ui/layouts/MobileApp'
 
 export default function QRCodePage() {
   const videoEl = React.useRef<HTMLVideoElement>()
 
+  function appendUrlParams(url: string): any {
+    const params = new URLSearchParams(new URL(window.location.href).search)
+    for (const [key, value] of params.entries()) {
+      url = url.concat(`&${key}=${value}`)
+    }
+    return url
```

Comment:
you don't have to manually `concat`. You can directly access the `searchParams` of `url` and append to it:

```suggestion
    const params = new URLSearchParams(new URL(window.location.href).search)
    const qr_url = new Url(url)
    
    for (const [key, value] of params.entries()) {
      qr_url.searchParams.append(key, value)
    }
    return qr_url.toString()
```

Reply:
Hi @salimhb , I implemented your suggestions.